# Fix FTBFS in hurd-i386 regarding bad check of PATH_MAX.
# Closes: #378966
# Thanks to Cyril Brulebois <cyril.brulebois@enst-bretagne.fr>
diff -Nur lib/chdir-long.c lib/chdir-long.c
--- lib/chdir-long.c	2005-09-19 21:12:18.000000000 +0000
+++ lib/chdir-long.c	2006-07-20 02:47:32.000000000 +0000
@@ -39,9 +39,7 @@
 # define O_DIRECTORY 0
 #endif
 
-#ifndef PATH_MAX
-# error "compile this file only if your system defines PATH_MAX"
-#endif
+#ifdef PATH_MAX
 
 struct cd_buf
 {
@@ -269,6 +267,8 @@
 }
 #endif
 
+#endif /* PATH_MAX */
+
 /*
 Local Variables:
 compile-command: "gcc -DTEST_CHDIR=1 -DHAVE_CONFIG_H -I.. -g -O -W -Wall chdir-long.c libcoreutils.a"

